### PR TITLE
Add KB article on how to host static files

### PR DESCRIPTION
**Pull request by sfc-gh-jcarroll**
## 📚 Context

Add a KB article for static file serving (merged in https://github.com/streamlit/streamlit/pull/5691)

## 🧠 Description of Changes

- Add a knowledge base article for hosting static files
- Link it from the "Using Streamlit" index

**Revised:**

https://deploy-preview-586--streamlit-docs.netlify.app/knowledge-base/using-streamlit/how-host-static-files

**Current:**

Page does not exist

## 💥 Impact

We have docs for the new feature!

Size:

- [ ] Small 
- [x] Not small 

## 🌐 References

- [ ] [Notion](https://www.notion.so/streamlit/Ability-to-serve-static-files-88a97775bd6a49d5a71f190a8e3c7dbf)

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Comment by sfc-gh-jcarroll** [comment]
@kajarenc @sfc-gh-hpathak ^ draft documentation for static file serving. You can see the preview page linked in the description. Please review the content and make comments on anything we should change!

@snehankekre FYI. I wanted to get the content quickly, but not attached to specifics if you want to make it align more to the rest of the docs! And no big rush on this.

**Review by snehankekre — APPROVED** [review]
Woo! This looks great to me. I didn't expect such a quick turnaround and assumed this would be a fast-follow 😄 

We can merge unless @kajarenc @sfc-gh-hpathak have objections or further edits.

**Comment by sfc-gh-jcarroll** [comment]
Let's get an active approval from Karen and then I'm good to merge. If there's other feedback we can fix it in a later PR.

**Comment by sfc-gh-kjavadyan** [comment]
Looks absolutely great!!! ❤️ 

Maybe one thing that missing is explicit limits that are hardcoded (on the other hand, maybe we should not mention them directly to not provocate attempts to bypass it :) :)

**Review by kajarenc — APPROVED** [review]

**Comment by sfc-gh-jcarroll** [comment]
I agree Karen - let's leave it out for now and we can add it if we see users are running into those issues and giving feedback.

@snehankekre I think good to merge whenever it makes sense in relation to the release - I don't have merge access so whenever you want to 😄 thanks!

**Pull request merged**
